superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -539,32 +517,37 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
       if (validationErrors && !isEmpty(validationErrors)) {
         return;
       }
+      const parameters_schema = isEditMode
+        ? dbToUpdate.parameters_schema.properties
+        : dbModel?.parameters.properties;
+      const additionalEncryptedExtra = JSON.parse(
+        dbToUpdate.encrypted_extra || '{}',
+      );
+      const paramConfigArray = Object.keys(parameters_schema || {});
 
-      const engine = dbToUpdate.backend || dbToUpdate.engine;
-      if (engine === 'bigquery' && dbToUpdate.parameters?.credentials_info) {
-        // wrap encrypted_extra in credentials_info only for BigQuery
+      paramConfigArray.forEach(paramConfig => {
+        // we are going through the parameter configuration and seeing if this is an encrypted field
```

Comment:
A suggestion for comments: good comments describe **why** something is being done, instead of **what** is being done. Something like this:

```js
/* 
 * Parameters that are annotated with the `x-encrypted-extra` properties should be moved to
 * `encrypted_extra`, so that they are stored encrypted in the backend when the database is
 * created or edited.
 */
```